src/types/type_param.rs

```diff
@@ -56,6 +56,19 @@ pub enum TypeArg {
     CustomValue(serde_yaml::Value),
 }
 
+impl TypeArg {
+    /// Report [`TypeArg`] if param is a type
+    pub fn tag(&self) -> Option<TypeTag> {
```

Comment:
Might be better to rename this, it's a bit different to `PrimType::tag`. Maybe `tag_if_type` (or `tag_of_type`) ?